<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="20"/>
  <arg name="map_size_y" default="30"/>
  <arg name="map_size_z" default="5"/>

  <arg name="c_num" default="0.0"/>
  <arg name="p_num" default="150.0"/>


  <node pkg ="map_generator" name ="random_forest" type ="random_forest" output = "screen">    
    <!-- this is the map init position, not the odom-->
    <param name="init_state_x"   value="0.0"/>
    <param name="init_state_y"   value="0.0"/>
    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/resolution" value="0.1"/>        
    <param name="map/frame_id"     value="$(arg map_frame_id)" />

    <param name="ObstacleShape/seed" value="-1"/>
    <param name="map/obs_num"    value="$(arg p_num)"/>
    <param name="ObstacleShape/lower_rad" value="0.3"/>
    <param name="ObstacleShape/upper_rad" value="1.0"/>
    <param name="ObstacleShape/lower_hei" value="5.0"/>
    <param name="ObstacleShape/upper_hei" value="5.0"/>        
    <param name="ObstacleShape/set_semantics" value="true"/>     

    <param name="map/circle_num" value="$(arg c_num)"/>        
    <param name="ObstacleShape/radius_l" value="0.7"/>        
    <param name="ObstacleShape/radius_h" value="0.5"/>        
    <param name="ObstacleShape/z_l" value="0.7"/>        
    <param name="ObstacleShape/z_h" value="0.8"/>        
    <param name="ObstacleShape/theta" value="0.5"/>        

    <param name="sensing/radius" value="5.0"/>        
    <param name="sensing/rate"   value="10.0"/>        
  </node>

</launch>